core/src/main/java/org/infinispan/interceptors/impl/CacheLoaderInterceptor.java

```diff
@@ -491,23 +455,21 @@ public WrappedEntrySet(EntrySetCommand command, CacheSet<CacheEntry<K, V>> entry
          this.entrySet = entrySet;
       }
 
-      long calculateTimeoutSeconds() {
-         int minimum = 10;
-         int size = dataContainer.sizeIncludingExpired();
-         if (size < 10_000) return minimum;
-         return Math.round(Math.log1p(size) * 10);
-      }
-
       @Override
       public CloseableIterator<CacheEntry<K, V>> iterator() {
-         CloseableIterator<CacheEntry<K, V>> iterator = Closeables.iterator(entrySet.stream());
+         // This can be a HashSet since it is only written to from the local iterator which is only invoked
+         // from user thread
          Set<K> seenKeys = new HashSet<>(cache.getAdvancedCache().getDataContainer().sizeIncludingExpired());
```

Comment:
1. I don't know if the listener will work very well. I will have to listen to all modifications to the cache as well all activations and passivations. My thought for fixing this was to have it store all if it is OBJECT storage and only a subset if it is OFF-HEAP or BINARY.

2. True about the injection, I can do that.

Reply:
Also a side note but after I add in segmented store and data container, this code will only be used for non dist stores. And it is possible with other cache modes to do iteration over each segment individually so that you only have so many resurrected objects.

But I will think more on doing a live view, it gets a bit messy with the listeners.

https://github.com/wburns/infinispan/blob/ISPN-8905_non_shared_store_segments/core/src/main/java/org/infinispan/stream/impl/local/PersistencKeyStreamSupplier.java#L78